### PR TITLE
Port and enable tutorial `04-low-memory-dropout`

### PR DESCRIPTION
**Pull request by prathams417**
Also comment out tutorial `03-matrix-multiplication` due to failure. `test-triton.sh` updated to reflect changes.

### CODE REVIEW

**Review by etiotto — COMMENTED** [review]
Pls add the new tutorial test to the CI yaml file (build_and_test.yml) as well.

File: scripts/test-triton.sh [hunk 1]
```diff
@@ -166,7 +166,8 @@ function run_tutorial_tests {
 
   run_tutorial_test "01-vector-add" 01-vector-add.py
   run_tutorial_test "02-fused-softmax" 02-fused-softmax.py
-  run_tutorial_test "03-matrix-multiplication" 03-matrix-multiplication.py
+  # run_tutorial_test "03-matrix-multiplication" 03-matrix-multiplication.py Currently failing
```
Comment (whitneywhtsang) [inline scripts/test-triton.sh:169]: Let's just remove it.

**Review by whitneywhtsang — COMMENTED** [review]
Please also add it to build_and_test.yml to prevent regression.

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -198,6 +198,7 @@ jobs:
           cd python/tutorials
           python3 01-vector-add.py
           python3 02-fused-softmax.py
+          python3 04-low-memory-dropout
```
Comment (whitneywhtsang) [inline .github/workflows/build_and_test.yml:201]:
```suggestion
          python3 04-low-memory-dropout.py
```

**Review by whitneywhtsang — APPROVED** [review]

File: .github/workflows/build_and_test.yml [hunk 3]
```diff
@@ -198,6 +198,7 @@ jobs:
           cd python/tutorials
```
Comment (whitneywhtsang) [inline .github/workflows/build_and_test.yml:198]: Need to also install `tabulate` on line 197.

**Pull request merged**